staticAssert statement

stepancheg opened the pull request:
`staticAssert` statement is a better replacement of `StaticAssert` function.
`staticAssert` can be used as top level statement.

Example:

```
record MySequenceImpl ( ... );
...
staticAssert(Sequence?(MySequenceImpl));
```

Pull request includes two more patches:
- remove StaticAssert function
- verify some types in io module types with staticAssert statements (as a demonstration)

stepancheg committed:
kill old StaticAssert function

stepancheg committed:
use staticassert to verify protocols in io.* modules

Verify that InputStream and OutputStream implementations really
conform corresponding protocol.

jckarter commented:
Looks good. I think it'd be more consistent if the keyword name were all-lowercase `staticassert`. It'd also be cool if `staticassert` could take a list of static values to compose into an error message, similar to how `eval` takes a list of static values and composes them into a source string. Then you could do things like `staticassert(TypeSize(T) < N, "T should be smaller than ", #N, " bytes but is ", #TypeSize(T), " bytes")`

stepancheg committed:
staticassert statement

`staticassert` is a better replacement of `StaticAssert` function.
`staticassert` can be used as top level statement.

Example:

```
record MySequenceImpl ( ... );
...
staticassert(Sequence?(MySequenceImpl));
```

`staticassert` has optional message parameters:

```
staticassert(TypeSize(T) > N, "type ", T ", " size must be at least ", N, " bytes");
```

stepancheg commented:
@jckarter done
- renamed `staticAssert` to `staticassert`
- added optional message varargs to `staticassert` statement

jckarter commented:
Thanks @stepancheg .
